mesonbuild/compilers/clike.py

```diff
@@ -384,7 +384,7 @@ def _get_basic_compiler_args(self, env, mode):
         # Select a CRT if needed since we're linking
         if mode == 'link':
             args += self.get_linker_debug_crt_args()
-        if mode in {'compile', 'preprocess'}:
+        if mode in {'compile', 'preprocess', 'link'}:
```

Comment:
the `elif mode == 'link':` below is now dead code. Relatedly, is `mode in {'compile', 'preprocess', 'link'}:` every not true?

Reply:
I think this is the problem, drop the `if mode in ...` check and make that code unconditional, and then replace the `elif` with `if`.